rclpy/src/rclpy/_rclpy_logging.cpp

```diff
@@ -63,15 +67,21 @@ rclpy_logging_shutdown()
  *
  * \param[in] name Fully-qualified name of logger.
  * \param[in] level to set
+ * \param[in] detailed_error True for reporting detailed rcutils error.
  * \return None
  */
 void
-rclpy_logging_set_logger_level(const char * name, int level)
+rclpy_logging_set_logger_level(const char * name, int level, bool detailed_error = false)
 {
+  std::lock_guard<std::mutex> lock(g_logging_lock);
```

Comment:
this global mutex should be locked here until we develop the mutex lock in rcutils level to protect the hash map.

Reply:
Yes.